### ISSUE TITLE
`undefined method 'space_before?' for an instance of Array` with prism parser

### ISSUE THREAD

**Issue by jdelStrother** [rubocop/rubocop#13820]
When using `ParserEngine: parser_prism`, the Layout/SpaceInsideReferenceBrackets cop fails on this ruby code:

```ruby
module Test
  A = /([^＠＃])/
  B = /
    #{foo[:bar]}
  /x
end
```

(note that those are unicode-y versions of `@` & `#` in the first regex, which appears to be part of the problem)

--------

## Expected behavior

Successfully parses the file

## Actual behavior

```
configuration from /Users/jon/Developer/web/.rubocop.yml
Default configuration from /Users/jon/Developer/web/vendor/bundle/ruby/3.3.0/gems/rubocop-1.71.2/config/default.yml
Inspecting 1 file
Scanning /Users/jon/Developer/web/test.rb
An error occurred while Layout/SpaceInsideReferenceBrackets cop was inspecting /Users/jon/Developer/web/test.rb:4:6.
undefined method `space_before?' for an instance of Array
/Users/jon/Developer/web/vendor/bundle/ruby/3.3.0/gems/rubocop-1.71.2/lib/rubocop/cop/mixin/surrounding_space.rb:74:in `extra_space?'
/Users/jon/Developer/web/vendor/bundle/ruby/3.3.0/gems/rubocop-1.71.2/lib/rubocop/cop/mixin/surrounding_space.rb:47:in `no_space_offenses'
/Users/jon/Developer/web/vendor/bundle/ruby/3.3.0/gems/rubocop-1.71.2/lib/rubocop/cop/layout/space_inside_reference_brackets.rb:84:in `on_send'
/Users/jon/Developer/web/vendor/bundle/ruby/3.3.0/gems/rubocop-1.71.2/lib/rubocop/cop/commissioner.rb:143:in `public_send'
/Users/jon/Developer/web/vendor/bundle/ruby/3.3.0/gems/rubocop-1.71.2/lib/rubocop/cop/commissioner.rb:143:in `block (2 levels) in trigger_restricted_cops'
/Users/jon/Developer/web/vendor/bundle/ruby/3.3.0/gems/rubocop-1.71.2/lib/rubocop/cop/commissioner.rb:171:in `with_cop_error_handling'
/Users/jon/Developer/web/vendor/bundle/ruby/3.3.0/gems/rubocop-1.71.2/lib/rubocop/cop/commissioner.rb:142:in `block in trigger_restricted_cops'
/Users/jon/Developer/web/vendor/bundle/ruby/3.3.0/gems/rubocop-1.71.2/lib/rubocop/cop/commissioner.rb:141:in `each'
/Users/jon/Developer/web/vendor/bundle/ruby/3.3.0/gems/rubocop-1.71.2/lib/rubocop/cop/commissioner.rb:141:in `trigger_restricted_cops'
/Users/jon/Developer/web/vendor/bundle/ruby/3.3.0/gems/rubocop-1.71.2/lib/rubocop/cop/commissioner.rb:70:in `on_send'
/Users/jon/Developer/web/vendor/bundle/ruby/3.3.0/gems/rubocop-ast-1.38.0/lib/rubocop/ast/traversal.rb:137:in `block in on_dstr'
/Users/jon/Developer/web/vendor/bundle/ruby/3.3.0/gems/rubocop-ast-1.38.0/lib/rubocop/ast/traversal.rb:137:in `each'
/Users/jon/Developer/web/vendor/bundle/ruby/3.3.0/gems/rubocop-ast-1.38.0/lib/rubocop/ast/traversal.rb:137:in `on_dstr'
/Users/jon/Developer/web/vendor/bundle/ruby/3.3.0/gems/rubocop-1.71.2/lib/rubocop/cop/commissioner.rb:71:in `on_begin'
/Users/jon/Developer/web/vendor/bundle/ruby/3.3.0/gems/rubocop-ast-1.38.0/lib/rubocop/ast/traversal.rb:137:in `block in on_dstr'
/Users/jon/Developer/web/vendor/bundle/ruby/3.3.0/gems/rubocop-ast-1.38.0/lib/rubocop/ast/traversal.rb:137:in `each'
/Users/jon/Developer/web/vendor/bundle/ruby/3.3.0/gems/rubocop-ast-1.38.0/lib/rubocop/ast/traversal.rb:137:in `on_dstr'
/Users/jon/Developer/web/vendor/bundle/ruby/3.3.0/gems/rubocop-1.71.2/lib/rubocop/cop/commissioner.rb:71:in `on_regexp'
/Users/jon/Developer/web/vendor/bundle/ruby/3.3.0/gems/rubocop-ast-1.38.0/lib/rubocop/ast/traversal.rb:152:in `on_casgn'
/Users/jon/Developer/web/vendor/bundle/ruby/3.3.0/gems/rubocop-1.71.2/lib/rubocop/cop/commissioner.rb:71:in `on_casgn'
/Users/jon/Developer/web/vendor/bundle/ruby/3.3.0/gems/rubocop-ast-1.38.0/lib/rubocop/ast/traversal.rb:137:in `block in on_dstr'
/Users/jon/Developer/web/vendor/bundle/ruby/3.3.0/gems/rubocop-ast-1.38.0/lib/rubocop/ast/traversal.rb:137:in `each'
/Users/jon/Developer/web/vendor/bundle/ruby/3.3.0/gems/rubocop-ast-1.38.0/lib/rubocop/ast/traversal.rb:137:in `on_dstr'
/Users/jon/Developer/web/vendor/bundle/ruby/3.3.0/gems/rubocop-1.71.2/lib/rubocop/cop/commissioner.rb:71:in `on_begin'
/Users/jon/Developer/web/vendor/bundle/ruby/3.3.0/gems/rubocop-ast-1.38.0/lib/rubocop/ast/traversal.rb:138:in `on_while'
/Users/jon/Developer/web/vendor/bundle/ruby/3.3.0/gems/rubocop-1.71.2/lib/rubocop/cop/commissioner.rb:71:in `on_module'
/Users/jon/Developer/web/vendor/bundle/ruby/3.3.0/gems/rubocop-ast-1.38.0/lib/rubocop/ast/traversal.rb:20:in `walk'
/Users/jon/Developer/web/vendor/bundle/ruby/3.3.0/gems/rubocop-1.71.2/lib/rubocop/cop/commissioner.rb:87:in `investigate'
/Users/jon/Developer/web/vendor/bundle/ruby/3.3.0/gems/rubocop-1.71.2/lib/rubocop/cop/team.rb:174:in `investigate_partial'
/Users/jon/Developer/web/vendor/bundle/ruby/3.3.0/gems/rubocop-1.71.2/lib/rubocop/cop/team.rb:108:in `investigate'
/Users/jon/Developer/web/vendor/bundle/ruby/3.3.0/gems/rubocop-1.71.2/lib/rubocop/runner.rb:348:in `block in inspect_file'
/Users/jon/Developer/web/vendor/bundle/ruby/3.3.0/gems/rubocop-1.71.2/lib/rubocop/runner.rb:347:in `each'
/Users/jon/Developer/web/vendor/bundle/ruby/3.3.0/gems/rubocop-1.71.2/lib/rubocop/runner.rb:347:in `flat_map'
/Users/jon/Developer/web/vendor/bundle/ruby/3.3.0/gems/rubocop-1.71.2/lib/rubocop/runner.rb:347:in `inspect_file'
/Users/jon/Developer/web/vendor/bundle/ruby/3.3.0/gems/rubocop-1.71.2/lib/rubocop/runner.rb:290:in `block in do_inspection_loop'
/Users/jon/Developer/web/vendor/bundle/ruby/3.3.0/gems/rubocop-1.71.2/lib/rubocop/runner.rb:324:in `block in iterate_until_no_changes'
<internal:kernel>:187:in `loop'
/Users/jon/Developer/web/vendor/bundle/ruby/3.3.0/gems/rubocop-1.71.2/lib/rubocop/runner.rb:317:in `iterate_until_no_changes'
/Users/jon/Developer/web/vendor/bundle/ruby/3.3.0/gems/rubocop-1.71.2/lib/rubocop/runner.rb:286:in `do_inspection_loop'
/Users/jon/Developer/web/vendor/bundle/ruby/3.3.0/gems/rubocop-1.71.2/lib/rubocop/runner.rb:167:in `block in file_offenses'
/Users/jon/Developer/web/vendor/bundle/ruby/3.3.0/gems/rubocop-1.71.2/lib/rubocop/runner.rb:192:in `file_offense_cache'
/Users/jon/Developer/web/vendor/bundle/ruby/3.3.0/gems/rubocop-1.71.2/lib/rubocop/runner.rb:166:in `file_offenses'
/Users/jon/Developer/web/vendor/bundle/ruby/3.3.0/gems/rubocop-1.71.2/lib/rubocop/runner.rb:154:in `process_file'
/Users/jon/Developer/web/vendor/bundle/ruby/3.3.0/gems/rubocop-1.71.2/lib/rubocop/runner.rb:135:in `block in each_inspected_file'
/Users/jon/Developer/web/vendor/bundle/ruby/3.3.0/gems/rubocop-1.71.2/lib/rubocop/runner.rb:134:in `each'
/Users/jon/Developer/web/vendor/bundle/ruby/3.3.0/gems/rubocop-1.71.2/lib/rubocop/runner.rb:134:in `reduce'
/Users/jon/Developer/web/vendor/bundle/ruby/3.3.0/gems/rubocop-1.71.2/lib/rubocop/runner.rb:134:in `each_inspected_file'
/Users/jon/Developer/web/vendor/bundle/ruby/3.3.0/gems/rubocop-1.71.2/lib/rubocop/runner.rb:120:in `inspect_files'
/Users/jon/Developer/web/vendor/bundle/ruby/3.3.0/gems/rubocop-1.71.2/lib/rubocop/runner.rb:73:in `run'
/Users/jon/Developer/web/vendor/bundle/ruby/3.3.0/gems/rubocop-1.71.2/lib/rubocop/cli/command/execute_runner.rb:26:in `block in execute_runner'
/Users/jon/Developer/web/vendor/bundle/ruby/3.3.0/gems/rubocop-1.71.2/lib/rubocop/cli/command/execute_runner.rb:52:in `with_redirect'
/Users/jon/Developer/web/vendor/bundle/ruby/3.3.0/gems/rubocop-1.71.2/lib/rubocop/cli/command/execute_runner.rb:25:in `execute_runner'
/Users/jon/Developer/web/vendor/bundle/ruby/3.3.0/gems/rubocop-1.71.2/lib/rubocop/cli/command/execute_runner.rb:17:in `run'
/Users/jon/Developer/web/vendor/bundle/ruby/3.3.0/gems/rubocop-1.71.2/lib/rubocop/cli/command.rb:11:in `run'
/Users/jon/Developer/web/vendor/bundle/ruby/3.3.0/gems/rubocop-1.71.2/lib/rubocop/cli/environment.rb:18:in `run'
/Users/jon/Developer/web/vendor/bundle/ruby/3.3.0/gems/rubocop-1.71.2/lib/rubocop/cli.rb:122:in `run_command'
/Users/jon/Developer/web/vendor/bundle/ruby/3.3.0/gems/rubocop-1.71.2/lib/rubocop/cli.rb:129:in `execute_runners'
/Users/jon/Developer/web/vendor/bundle/ruby/3.3.0/gems/rubocop-1.71.2/lib/rubocop/cli.rb:51:in `block in run'
/Users/jon/Developer/web/vendor/bundle/ruby/3.3.0/gems/rubocop-1.71.2/lib/rubocop/cli.rb:81:in `profile_if_needed'
/Users/jon/Developer/web/vendor/bundle/ruby/3.3.0/gems/rubocop-1.71.2/lib/rubocop/cli.rb:43:in `run'
/Users/jon/Developer/web/vendor/bundle/ruby/3.3.0/gems/rubocop-1.71.2/exe/rubocop:19:in `<top (required)>'
bin/rubocop:8:in `load'
bin/rubocop:8:in `<main>'
C

Offenses:

test.rb:1:1: C: [Correctable] Style/FrozenStringLiteralComment: Missing frozen string literal comment.
module Test
^

1 file inspected, 1 offense detected, 1 offense autocorrectable

1 error occurred:
An error occurred while Layout/SpaceInsideReferenceBrackets cop was inspecting /Users/jon/Developer/web/test.rb:4:6.
configuration from /Users/jon/Developer/web/.rubocop.yml
Errors are usually caused by RuboCop bugs.
Please, update to the latest RuboCop version if not already in use, and report a bug if the issue still occurs on this version.
https://github.com/rubocop/rubocop/issues

Mention the following information in the issue report:
1.71.2 (using Prism 1.3.0, rubocop-ast 1.38.0, analyzing as Ruby 3.3, running on ruby 3.3.6) [arm64-darwin24]
Finished in 0.10235799999645678 seconds
```

## Steps to reproduce the problem

* create test.rb:
```ruby
module Test
  A = /([^＠＃])/
  B = /
    #{foo[:bar]}
  /x
end
```
* create .rubocop.yml:
```yml
AllCops:
  ParserEngine:  parser_prism
  TargetRubyVersion: 3.3
```
* `rubocop test.rb`

## RuboCop version

```
$  rubocop -V
1.71.2 (using Prism 1.3.0, rubocop-ast 1.38.0, analyzing as Ruby 3.3, running on ruby 3.3.6) [arm64-darwin24]
```

**Comment by Earlopain** [comment]
Thanks for the report! This will be fixed in the next prism release.

I can't say which PR specifically fixed this, the error doesn't look familiar, but it does work on current main. The closest one would be https://github.com/ruby/prism/pull/3400, though that was only for multiline strings with multibyte characters (maybe regex always go through that codepath).

**Comment by jdelStrother** [comment]
Oh, awesome - thanks for trying that.

**Comment by dvandersluis** [comment]
Prism 1.4.0 has been released which should fix this issue. Please upgrade the gem and try again.
